### PR TITLE
Add defined? -> instance_variable_defined? mutation

### LINKED ISSUE #1138: Add mutation from `defined(@ivar)` to `instance_variable_defined?(:ivar)`.

### PR DESCRIPTION
**Pull request by dgollahon**
- Mutates expressions where `defined?` is being used to check specifically for an instance variable such as `defined?(@a)` to the more narrow `instance_variable_defined?(:@a)` instead.
- Closes #1138.

### CODE REVIEW

File: meta/defined.rb [hunk 1]
```diff
@@ -7,3 +7,13 @@
...
   mutation 'true'
 end
+
+Mutant::Meta::Example.add :defined? do
+  source 'defined?(@foo)'
+
+  singleton_mutations
+  mutation 'defined?(nil)'
+  mutation 'defined?(foo)'
+  mutation 'true'
```
Comment (dgollahon) [inline meta/defined.rb:17]: I would do this as a separate PR, but should the `defined?` mutator also emit `false`? It's interesting that it is only true. Is there a reason for that?

File: Changelog.md [hunk 2]
```diff
@@ -20,8 +24,8 @@
     This is useful when reacting to noop errors.
 
 * [#1154](https://github.com/mbj/mutant/pulls/1154)
- 
-  * Add subcommand `environment subject list`. It allows to list 
+
+  * Add subcommand `environment subject list`. It allows to list
```
Comment (dgollahon) [inline Changelog.md:28]: My editor trimmed some trailing whitespace. I can remove this from the commit if you want but also seems like it is worth fixing.

File: meta/defined.rb (hunk 1 above)
Comment (mbj) [reply meta/defined.rb]: I think this was a deliberate decision as `singleton_mutations` never emits `false`:

```
irb(main):001:0> defined?(@a)
=> nil
irb(main):002:0> @a = "foo"
=> "foo"
irb(main):003:0> defined?(@a)
=> "instance-variable"
```

emitting `true` is also wrong in that light ;) I'll add a commit to improve this.

File: Changelog.md (hunk 2 above)
Comment (mbj) [reply Changelog.md:36]: Added an extra commit, and my editors whitespace trimming broke ;) Good catch.

File: meta/defined.rb (hunk 1 above)
Comment (dgollahon) [reply meta/defined.rb]: Ah, I guess since we are emitting `nil` we already have something falsey.

I actually think `true` was a good mutation--don't we normally replace predicates with `true` and `false`? (or maybe there is an issue for this somewhere.

Or are you thinking it's just too hard to cover normal `defined?` mutations because it's hard to have both states?

Comment (dgollahon) [reply meta/defined.rb]: oh! nevermind. I see that `defined?` never returns `true` normally. Ok, cool, good call, ignore me. I misunderstood what you were saying for a moment. Nice improvement.

Comment (mbj) [reply meta/defined.rb]: no worries, `defined?` is weird in other ways.

It actually gives information about the AST, but when the AST references a constant, than it also checks the presence of that constant. Ruby....
